### DISCUSSION TITLE
whether langium can support specified langage lexer and parse validate

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1416] [Q&A] [answered]
also,recently,there is a new requirement,whether langium can support specific langage lexer and parse validate ,for example,plc st(Structured text)langauge

**Comment by msujew** [comment]
@zrwsmd I can't tell you much, but I can confirm that Structured Text is parsable with Langium. Note that Langium supports all languages that can be expressed in [LL](https://en.m.wikipedia.org/wiki/LL_parser).

**Reply by zrwsmd** [reply]
i also think it can be achieved by langium,but in that case i might write st.langium by myself,i mean whether langium have ready-made 
achievement for this specific langage

**Reply by msujew** [reply, marked as answer]
You might interested in this PR: https://github.com/Serhioromano/vscode-st/pull/51. Aside from that, I'm not sure there's another open source implementation of the language.
